### PR TITLE
Implemented Doctor management API with value and reference types

### PR DESCRIPTION
**Pull request by mseheda**
- Added Person and Doctor entities with private fields, public properties, and methods.
- Implemented DoctorService and IDoctorService for handling in-memory storage of doctors.
- Created DoctorsController to expose API endpoints for managing doctors.
- Included example of boxing/unboxing and usage of object type.
- Overrode Equals() and GetHashCode() methods in Doctor for object comparison.
- Configured Startup.cs to include necessary services and Swagger for API documentation.
- API successfully tested through both GUI and terminal with POST and GET requests.

### CODE REVIEW

File: Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/obj/Debug/net8.0/Healthcare-Hospital-Management-System.AssemblyInfo.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@
 [assembly: System.Reflection.AssemblyCompanyAttribute("Healthcare-Hospital-Management-System")]
```
Comment (liudmyla-zinchenko) [inline Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/obj/Debug/net8.0/Healthcare-Hospital-Management-System.AssemblyInfo.cs]: Required: remove bin and obj  folder, add them to gitignore, delete from repository

**Review by liudmyla-zinchenko — CHANGES_REQUESTED** [review]
1 required change: remove folders bin and obj and add them to gitingnore.
After fixing this and merging 1 PR I will approve this one

**Comment by mseheda** [comment]
Removed unneccessary files and folders.

**Review by liudmyla-zinchenko — APPROVED** [review]
Accepted

**Pull request merged**